src/base/AsyncJob.cc

```diff
@@ -12,14 +12,21 @@
 #include "base/AsyncCall.h"
 #include "base/AsyncJob.h"
 #include "base/AsyncJobCalls.h"
+#include "base/PackableStream.h"
 #include "base/TextException.h"
 #include "cbdata.h"
+#include "mem/PoolingAllocator.h"
 #include "MemBuf.h"
+#include "mgr/Registration.h"
+#include "Store.h"
 
 #include <ostream>
+#include <unordered_set>
 
 InstanceIdDefinitions(AsyncJob, "job");
 
+static std::unordered_set<AsyncJob *, std::hash<AsyncJob *>, std::equal_to<AsyncJob *>, PoolingAllocator<AsyncJob *> > Jobs_;
```

Comment:
Please do not use global statics. Correct initialization keeps coming up as problematic for these.
A variable which is only used internally by the `class AsyncJob` can be a private member, otherwise prefer to wrap with a function for cross-unit access.

Reply:
Update: developer meeting today decided to start a policy of "no new globals".

Reply:
Update: developer meeting today decided to start a policy of "no new globals".

Reply:
Branch commit b05b183 adheres to the new "no new globals" policy.

Reply:
The documentation for the new "no new globals" policy has been proposed at https://github.com/squid-cache/squid-cache.github.io/pull/6. The code in that documentation proposal matches this PR code.